### ISSUE TITLE
xbuild fails on fresh git clone

### ISSUE THREAD

**Issue by nii236** [QuantConnect/Lean#149]
Hey all,

I did a fresh `git clone` and ran `xbuild` (from OSX). This is the error that pops up:

```
/Users/nii236/git/lean-ref/QuantConnect.Lean.sln:  warning : /Users/nii236/git/lean-ref/Algorithm.FSharp/QuantConnect.Algorithm.FSharp.fsproj: The required attribute "Project" in Import is empty
Project "/Users/nii236/git/lean-ref/QuantConnect.Lean.sln" (default target(s)):
	Target ValidateSolutionConfiguration:
		Building solution configuration "Debug|Any CPU".
	Target Build:
		Project "/Users/nii236/git/lean-ref/Logging/QuantConnect.Logging.csproj" (default target(s)):
			Target PrepareForBuild:
				Configuration: Debug Platform: AnyCPU
			Target GenerateSatelliteAssemblies:
			No input files were specified for target GenerateSatelliteAssemblies, skipping.
			Target GenerateTargetFrameworkMonikerAttribute:
			Skipping target "GenerateTargetFrameworkMonikerAttribute" because its outputs are up-to-date.
			Target CoreCompile:
			Skipping target "CoreCompile" because its outputs are up-to-date.
		Done building project "/Users/nii236/git/lean-ref/Logging/QuantConnect.Logging.csproj".
/Users/nii236/git/lean-ref/Algorithm.FSharp/QuantConnect.Algorithm.FSharp.fsproj: error : /Users/nii236/git/lean-ref/Algorithm.FSharp/QuantConnect.Algorithm.FSharp.fsproj: The required attribute "Project" in Import is empty
	Task "MSBuild" execution -- FAILED
	Done building target "Build" in project "/Users/nii236/git/lean-ref/QuantConnect.Lean.sln".-- FAILED
Done building project "/Users/nii236/git/lean-ref/QuantConnect.Lean.sln".-- FAILED

Build FAILED.

Warnings:

/Users/nii236/git/lean-ref/QuantConnect.Lean.sln:  warning : /Users/nii236/git/lean-ref/Algorithm.FSharp/QuantConnect.Algorithm.FSharp.fsproj: The required attribute "Project" in Import is empty

Errors:

/Users/nii236/git/lean-ref/QuantConnect.Lean.sln (default targets) ->
(Build target) ->

	/Users/nii236/git/lean-ref/Algorithm.FSharp/QuantConnect.Algorithm.FSharp.fsproj: error : /Users/nii236/git/lean-ref/Algorithm.FSharp/QuantConnect.Algorithm.FSharp.fsproj: The required attribute "Project" in Import is empty

	 1 Warning(s)
	 1 Error(s)
```

I'm not too sure what's going on here. My mono is of the latest version (4.0.2.5). I think the offending line *might* be the following in the fsharp file:

`<Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />`

I don't have `MSBuild`, I'm running `xbuild`.

Any ideas?

**Comment by mchandschuh** [comment]
Do you have the required FSharp dependencies installed? I don't have OSX, but I know on windows we need to install FSharp. If that doesn't work, a hack would be to set the QuantConnect.Algorithm.FSharp project to not build and remove it as a dependency from other projects.

**Comment by nii236** [comment]
For Debian based distros, installing `fsharp` separately to Mono worked. Not sure about OS X but I resolved it by removing the relevant projects that used F#.

**Comment by jaredbroad** [comment]
Please submit a read me pull request for readme files in the visual basic
project, we need the 'apt-get install mono-vbnc' for linux os,

I don't think mono supports vb 4.5,

Check out the java readme for an example:
https://github.com/QuantConnect/Lean/blob/master/Algorithm.Java/readme.md

On Thu, Sep 3, 2015 at 11:08 PM, John Nguyen <notifications@github.com>
wrote:

> Closed #149 <https://github.com/QuantConnect/Lean/issues/149>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/QuantConnect/Lean/issues/149#event-400767411>.
>



-- 
Jared Broad

www.quantconnect.com
Phone-USA: +1 917 327 0556
*Democratizing Finance, Empowering Individuals*
Facebook <http://www.facebook.com/QuantConnect> | Twitter
<https://twitter.com/#!/QuantConnect> | LinkedIn
<http://www.linkedin.com/company/quantconnect> | Skype: jaredbroad

**Comment by AnObfuscator** [comment]
Yeah, the VB Project's TargetFrameworkVersion was the culprit. Setting it to 4.0 allowed the solution to build.
